src/display/text_layer.js

```diff
@@ -142,12 +159,21 @@ function appendText(task, geom, styles, ctx) {
     left = tx[4] + fontAscent * Math.sin(angle);
     top = tx[5] - fontAscent * Math.cos(angle);
   }
+
+  const scaleFactorStr = "calc(var(--scale-factor)*";
+  const divStyle = textDiv.style;
   // Setting the style properties individually, rather than all at once,
   // should be OK since the `textDiv` isn't appended to the document yet.
-  textDiv.style.left = `${left}px`;
-  textDiv.style.top = `${top}px`;
-  textDiv.style.fontSize = `${fontHeight}px`;
-  textDiv.style.fontFamily = style.fontFamily;
+  if (task._container.tagName === "SPAN") {
```

Comment:
This would be safer as follows, since it's technically possible for a user to pass in a `span` as the original container.
```suggestion
  if (task._container.classList.contains("markedContent")) {
```

Reply:
The problem in general (regardless if we're in markedContent or not) is that using `%` will make the positions of the span we just created relative to its parent dimensions, and in the case of a marked content, the parent dimensions are 0.
So in general, if the parent is a span we mustn't use `%`.
Maybe we could use the `%` only when the parent is the text layer itself, it'd be likely better.

Reply:
> Maybe we could use the `%` only when the parent is the text layer itself, it'd be likely better.

Right, that might be easier/safer in general.